include/wlr/backend/session/interface.h

```diff
@@ -14,8 +14,8 @@
 struct session_impl {
 	struct wlr_session *(*create)(struct wl_display *disp);
 	void (*destroy)(struct wlr_session *session);
-	int (*open)(struct wlr_session *session, const char *path);
-	void (*close)(struct wlr_session *session, int fd);
+	int (*open)(struct wlr_session *session, const char *path, int *device_id);
```

Comment:
Why do we need an additional `device_id` arg? Can't the FD be used as an ID?

Reply:
Logind uses device major/minor for open, close, pause etc., which happens to be derivable from an FD, but these values does not have the same meaning on FreeBSD.

seatd uses filepath to open, and uses a returned ID for the rest. It would be nice if the FD could serve double-duty, but seatd can't know what FD will be assigned on the other end of the socket. Plus, the overloaded meaning might not make that much sense as it won't follow the file description.

Open to ideas.

Reply:
This question was more about the wlroots session interface, less about the seatd API.

It's easy for the libseat session implementation to keep a list of (FD, seatd device ID) if necessary. Session implementation details should not leak in the generic interface IMHO.

Reply:
Ah, sure. The wlroots libseat impl could definitely contain this for now. I just did it the easy way for testing.